extensions/arcnet/src/arc/net/dns/JndiContextNameserverProvider.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is a modified class of dnsjava, an implementation of the dns protocol in java.
+ * Licensed under the BSD-3-Clause.
+ */
+package arc.net.dns;
+
+import arc.struct.*;
+import arc.util.*;
+
+import javax.naming.*;
+import javax.naming.directory.*;
+import java.net.*;
+import java.util.*;
+
+import static arc.net.dns.ArcDns.dnsResolverPort;
+
+/**
+ * Resolver config provider that tries to extract the system's DNS servers from the
+ * <a href="https://docs.oracle.com/javase/8/docs/technotes/guides/jndi/jndi-dns.html">JNDI DNS Service Provider</a>.
+ */
+public final class JndiContextNameserverProvider implements NameserverProvider{
+    private InnerJndiContextNameserverProvider inner;
+
+    public JndiContextNameserverProvider(){
+        if(!OS.isAndroid){
+            try{
+                inner = new InnerJndiContextNameserverProvider();
+            }catch(NoClassDefFoundError e){
```

Comment:
Should catch a Throwable instead, just to be sure. I would not want a static initialization error here regardless of what it is.